File: 1863. Sum of All Subset XOR Totals.go
func subsetXORSum(nums []int) int {
	return sum(nums, 0, 0)
}
func sum(nums []int, index, current int) int {
	if index >= len(nums) {
		return current
	}
	return sum(nums, index+1, current) +
		sum(nums, index+1, nums[index]^current)
}

File: 1863. Sum of All Subset XOR Totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        return self.sum(nums, 0, 0)
    def sum(self, nums, index, current):
        if index >= len(nums): return current
        return self.sum(nums, index+1, current) + self.sum(nums, index+1, nums[index]^current)
    